File: Breadth_First_Search/BreadthFirstSearch.java
package Breadth_First_Search;

import java.util.*;

import _Graph.*;

public class BreadthFirstSearch {
    public List<Vertex> breadthFirstSearch(Graph G, Vertex src) {
        Set<Vertex> visited = new HashSet<>();
        Queue<Vertex> queue = new LinkedList<>();
        List<Vertex> result = new ArrayList<Vertex>();

        queue.offer(src);
        visited.add(src);
        while (!queue.isEmpty()) {
            Vertex vertex = queue.poll();
            result.add(vertex);
            for (Edge edge : G.getAdjacencyList().get(vertex)) {                
                Vertex neighbor = edge.v();
                if (!visited.contains(neighbor)) {
                    queue.offer(neighbor);
                    visited.add(neighbor);
                }
            }
        }
        return result;
    }
}

File: Breadth_First_Search/breadth_first_search.py
from .._Graph.graph import *
from queue import Queue

def breadth_first_search(G: Graph, src: Vertex) -> list[Vertex]:
    visited: set[Vertex] = set()
    queue: Queue[Vertex] = Queue()
    result: list[Vertex] = []

    queue.put(src)
    visited.add(src)
    while queue:
        vertex: Vertex = queue.get()
        result.append(vertex)
        for edge in G.get_adjacency_list()[vertex]:
            neighbor: Vertex = edge.v()
            if neighbor not in visited:
                queue.put(neighbor)
                visited.add(neighbor)
    return result
